### PR TITLE
Fix mishandling of NBSP with wrap turned on

### PR DESCRIPTION
**Pull request by pm100**
see issue #404

### CODE REVIEW

**Comment by fdehau** [comment]
Nice catch :). It seems we are missing a test on this case. Would you mind adding one ?

**Comment by pm100** [comment]
added test

passing comment. Its really a pain that you have termion as default backend, working on tui on windows is always v fiddly

File: tests/widgets_paragraph.rs [hunk 1]
```diff
@@ -141,6 +141,32 @@ fn widgets_paragraph_renders_mixed_width_graphemes() {
     terminal.backend().assert_buffer(&expected);
 }
 
+#[test]
+fn white_space_wrap(){
```
Comment (fdehau) [inline tests/widgets_paragraph.rs:145]:
```suggestion
fn widgets_paragraph_can_wrap_with_a_trailing_nbsp() {
```
to be consistent with other tests and quickly launch all tests linked to a widget :)

**Comment by fdehau** [comment]
> added test

Thanks :+1:  Rustfmt is not happy though.

> Its really a pain that you have termion as default backend, working on tui on windows is always v fiddly

I understand and `crossterm` will probably become the default backend at some point but I'm waiting for it to stabilize a bit.

**Pull request merged**